Domains/Jobs/Traits/Attributes.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @file Domains/Jobs/Traits/Attributes.php
```

Comment:
I'm still finding this trait a little extraneous; unless I'm reading it wrong, the defaults are always used, no? I'm also aware that you're introducing some new concepts here (e.g. `Eloquentable`) that should be considered for adoption elsewhere. To step back from this specific implementation, could you write a little about the new patterns here you'd like to propose for wider adoption and their relationship to existing concepts (e.g. `DataObject` vs. `Eloquentable`)?